src/libcore/macros.rs

```diff
@@ -445,9 +445,10 @@ macro_rules! writeln {
 /// * Iterators that dynamically terminate.
 ///
 /// If the determination that the code is unreachable proves incorrect, the
-/// program immediately terminates with a [`panic!`]. The function [`unreachable_unchecked`],
-/// which belongs to the [`std::hint`] module, informs the compiler to
-/// optimize the code out of the release version entirely.
+/// program immediately terminates with a [`panic!`].
+///
+/// The unsafe counterpart of this macro is the [`unreachable_unchecked`] function, which
+/// instead of a [`panic!`] will cause undefined behavior if the code is reached.
```

Comment:
```diff
-/// instead of a [`panic!`] will cause undefined behavior if the code is reached.
+/// will cause undefined behavior if the code is reached.
```

Reply:
I removed the phrase now as you suggest